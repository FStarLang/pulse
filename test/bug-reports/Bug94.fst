module Bug94

#lang-pulse
open Pulse

[@@expect_failure [19]]
fn foo ()
  requires emp
  ensures  emp
{
  let x : nat = -1;
  ()
}

[@@expect_failure [189]]
fn foo (y:string)
  requires emp
  ensures  emp
{
  let x : nat = y;
  ()
}

let divide
  (a: erased int)
  (b: erased int { reveal b <> 0 })
  : GTot (erased int) = a / b

[@@expect_failure [19]]
fn divide_test ()
  requires emp
  ensures emp
{
    let hundred = divide 100;
    ();
    let unknown = hundred 0;
    ()
}
