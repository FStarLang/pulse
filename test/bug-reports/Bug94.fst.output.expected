* Info at Bug94.fst(11,16-11,18):
  - Expected failure:
  - Ill-typed term: - 1
  - Expected a term of type Prims.nat
  - Assertion failed
  - The SMT solver could not prove the query. Use --query_stats for more
    details.
  - Also see: Prims.fst(682,18-682,24)

* Info at Bug94.fst(20,16-20,17):
  - Expected failure:
  - Expected expression of type Prims.nat got expression y of type Prims.string

* Info at Bug94.fst(34,18-34,24):
  - Expected failure:
  - Ill-typed term: hundred 0
  - Assertion failed
  - The SMT solver could not prove the query. Use --query_stats for more
    details.
  - See also Bug94.fst(26,19-26,32)

